Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -495,4 +545,31 @@ + (CDTQSqlParts *)selectStatementForAndClause:(NSArray *)clause usingIndex:(NSSt
     return parts;
 }
 
++ (CDTQSqlParts *)selectStatementForTextClause:(NSObject *)clause usingIndex:(NSString *)indexName
+{
+    if (!clause) {
+        return nil;  // no query here
+    }
+    
+    if (!indexName) {
+        return nil;
+    }
+    
+    if (![clause isKindOfClass:[NSDictionary class]]) {
+        return nil; // should never get here as this would not pass normalization
```

Comment:
A valid point @rhyshort.  I've changed the method to expect an NSDictionary instead of an NSObject and I've updated the calling code block to cast the clause as an NSDictionary.  Since this is how the other basic clauses are handled.  I've also added a comment there as you suggested.  Done in 66a688789ff1d951b6b921ba97852d52a6e0b903.